march_hardware/src/IMotionCube.cpp

```diff
@@ -364,6 +388,84 @@ uint32_t IMotionCube::computeSWCheckSum(int& start_address, int& end_address)
                                  delimiter);
 }
 
+bool IMotionCube::verifySetup()
+{
+  int start_address = 0, end_address = 0;
+  uint32_t sw_value = this->computeSWCheckSum(start_address, end_address);
+  // set parameters to compute checksum on the imc
+  int checksum_setup =
+      sdo_bit32_write(slaveIndex, 0x2069, 0, end_address * 65536 + start_address);  // Endaddress leftshifted 4 times
```

Comment:
```suggestion
      sdo_bit32_write(slaveIndex, 0x2069, 0, end_address << 16 + start_address);  // Endaddress leftshifted 4 times
```
I believe you can also do it something like this